aerogear-android-push/src/main/java/org/jboss/aerogear/android/unifiedpush/gcm/AeroGearGCMPushRegistrar.java

```diff
@@ -127,14 +129,9 @@ protected Exception doInBackground(Void... params) {
                     if (instanceId == null) {
                         instanceId = instanceIdProvider.get(context);
                     }
-                    String regid = getRegistrationId(context);
-
-                    if (regid.length() == 0) {
-                        regid = instanceId.getToken(senderId,
+                    String regid = instanceId.getToken(senderId,
```

Comment:
InstanceID gives you a registration Token.  It should probably be registrationToken.